modules/azure-storage-account/main.tf

```diff
@@ -0,0 +1,101 @@
+locals {
+  uses_cmk = var.customer_managed_key != null
+}
+
+resource "azurerm_storage_account" "storage_account" {
+  name                = var.name
+  location            = var.location
+  resource_group_name = var.resource_group_name
+
+  account_kind             = var.account_kind
+  account_tier             = var.account_tier
+  account_replication_type = var.account_replication_type
+  tags                     = var.tags
+
+  # secure by default
+  allow_nested_items_to_be_public = false
+  https_traffic_only_enabled      = true
+  min_tls_version                 = var.min_tls_version
+
+  # enable mounting account as disk
+  nfsv3_enabled  = var.is_nfs_mountable
+  is_hns_enabled = var.is_nfs_mountable
```

Comment:
tf docs : This can only be true when account_tier is Standard and account_kind is StorageV2, or account_tier is Premium and account_kind is BlockBlobStorage. Additionally, the is_hns_enabled is true and account_replication_type must be LRS or RAGRS.